resources/sample_vaults/Tasks-Demo/Filters/Regular Expression Searches.md

```diff
@@ -0,0 +1,238 @@
+### Be careful of special characters
+
+This search looks like it searches for `#context/pc_abigail`, but it actually searches for `#context`, because the `/` before `/pc` is interpreted as the end of the pattern being searched for.
+
+````text
+```tasks
+description regex matches /#context/pc_abigail/
+```
+````
+
+```tasks
+description regex matches /#context/pc_abigail/
+```
+
+Instead, `/` characters in the middle of the expression must be 'escaped' with a `\`:
+
+````text
+```tasks
+description regex matches /#context\/pc_abigail/
+```
+````
+
+```tasks
+description regex matches /#context\/pc_abigail/
+```
+
+### Tasks that to not match a regular expression
+
+````text
+```tasks
+description regex does not match /pc_abigail|pc_edwina|at_work/
+limit 5
+```
+````
+
+```tasks
+description regex does not match /pc_abigail|pc_edwina|at_work/
+limit 5
+```
+
+## Special characters
+
+### At least one digit in the description
+
+````text
+```tasks
+# Any digit
+description regex matches /\d/
+limit 5
+short mode
+```
+````
+
+```tasks
+# Any digit
+description regex matches /\d/
+limit 5
+short mode
+```
+
+### No digits in the description
+
+````text
+```tasks
+# Any nondigit
+description regex matches /^\D+$/
+limit 10
+short mode
+```
+````
+
+```tasks
+# Any nondigit
+description regex matches /^\D+$/
+limit 10
+short mode
+```
+
+## Invalid regex searches
+
+### Extra slashes truncate the query
+
+````text
+```tasks
+path regex matches /Filters/Regular Expression Searches/
+```
+````
+
+```tasks
+path regex matches /Filters/Regular Expression Searches/
+```
+
+Should give an error. Intention was to only match tasks in this file - but it matches all in this file.
+
+This is how to do the above search correctly, adding a `\` before each `/` inside the pattern:
+
+````text
+```tasks
+path regex matches /Filters\/Regular Expression Searches/
+```
+````
+
+```tasks
+path regex matches /Filters\/Regular Expression Searches/
+```
+
+### Missing `/`
+
+````text
+```tasks
+description regex matches CASE
+```
+````
+
+```tasks
+description regex matches CASE
+```
+
+Gives:
+`Tasks query: cannot parse regex (description); check your leading and trailing slashes for your query`
+
+### Invalid flag
+
+````text
+```tasks
+description regex matches /CASE/&
+```
+````
+
+```tasks
+description regex matches /CASE/&
+```
+
+Works. Should complain about invalid flag.
+
+### Mismatched square brackets
+
+````text
+```tasks
+description regex matches /[123/
+```
+````
+
+```tasks
+description regex matches /[123/
+```
+
+Gives:
+`Tasks query: cannot parse regex (description); check your leading and trailing slashes for your query`
+
+- [ ] Would like it to give a meaningful error message about what the regex error is.
+
+## Possible developer actions
+
+Intentionally without `#task` as not intended to be picked up by example searches above.
+
+- [ ] Work out how to prevent `path regex matches /a/b/c/d/` from confusingly only searching `path regex matches /a/`.
```

Comment:
Open an Issue against this?

Fixable with another regex. 😉😂🤣🤪

Reply:
Done. See #1037.